crates/sui-core/src/authority_server.rs

```diff
@@ -486,71 +558,86 @@ impl ValidatorService {
             // For shared objects this will wait until either timeout or we have heard back from consensus.
             // For owned objects this will return without waiting for certificate to be sequenced
             // First do quick dirty non-async check
-            if !epoch_store.is_tx_cert_consensus_message_processed(&certificate)? {
+            if !epoch_store.is_all_tx_certs_consensus_message_processed(&verified_certificates)? {
                 let _metrics_guard = if shared_object_tx {
                     Some(self.metrics.consensus_latency.start_timer())
                 } else {
                     None
                 };
-                let transaction = ConsensusTransaction::new_certificate_message(
-                    &self.state.name,
-                    certificate.clone().into(),
-                );
-                self.consensus_adapter.submit(
-                    transaction,
+                let transactions = verified_certificates
+                    .iter()
+                    .map(|certificate| {
+                        ConsensusTransaction::new_certificate_message(
+                            &self.state.name,
+                            certificate.clone().into(),
+                        )
+                    })
+                    .collect::<Vec<_>>();
+                self.consensus_adapter.submit_batch(
+                    &transactions,
```

Comment:
This seems like a potential bug, since we may be submitting already sequenced transactions, which would then be inserted into process_consensus_transactions. But it may be benign since insertions into that table are already racing with consensus handler. @mwtian any thoughts?

Reply:
My reading of the code is that this change does not seem to introduce new behavior. Transactions inflight or already sequenced in consensus can be submitted to consensus adapter, as mentioned by the "quick dirty non-async check" above. This behavior should be tolerated right now. I think only unique transactions by digests are or can be used for epoch change and revert.

Reply:
I agree - any extra checkpointed transactions in that table are ignored and not reverted